core/templates/pages/exploration-editor-page/translation-tab/state-translation/state-translation.component.ts

```diff
@@ -137,26 +141,38 @@ export class StateTranslationComponent implements OnInit, OnDestroy {
     return this.platformFeatureService.status.AddVoiceoverWithAccent.isEnabled;
   }
 
+  needsTranslation(): boolean {
+    return (
+      this.translationLanguageService.getActiveLanguageCode() !==
+      this.explorationLanguageCodeService.displayed
+    );
+  }
+
   getRequiredHtml(subtitledHtml: SubtitledHtml): string {
     if (this.translationTabActiveModeService.isTranslationModeActive()) {
       return subtitledHtml.html;
     }
 
     let langCode = this.translationLanguageService.getActiveLanguageCode();
+
+    if (langCode == this.explorationLanguageCodeService.displayed) {
+      return subtitledHtml.html;
+    }
+
     if (
       !this.entityTranslationsService.languageCodeToLatestEntityTranslations.hasOwnProperty(
         langCode
       )
     ) {
-      return subtitledHtml.html;
+      return null;
```

Comment:
Why did this (and the following) change?

Reply:
@sissamrcorreia Thanks for the explanation. This might be OK but it needs comments in the code, a docstring written for the function, and possibly a change to the name to more clearly indicate what the function is returning.

(I.e. I don't think the change is necessarily wrong -- it just needs to be communicated properly so that future developers reading this aren't confused. Thanks!)